libraries/botbuilder-core/src/showTypingMiddleware.ts

```diff
@@ -68,14 +68,15 @@ export class ShowTypingMiddleware implements Middleware {
             scheduleIndicator();
         }
 
-        // Execute remaining middleware, then clear scheduled indicators
-        await next();
-
-        finished = true;
-        if (timeout) clearTimeout(timeout);
+        // Execute remaining middleware inside try/finally to ensure we eventually clear timeouts
+        try {
+            await next();
+        } finally {
+            finished = true;
+            if (timeout) clearTimeout(timeout);
+        }
```

Comment:
Solid refactor.

Reply:
This was actually a bug fix. If an error is thrown the typing indicator is never cleared.

Reply:
Yes, I recognized the clearing of the timeout and didn’t type what I meant... 😫